src/mds/CInode.cc

```diff
@@ -4027,21 +4027,39 @@ void CInode::decode_import(bufferlist::const_iterator& p,
 
 void InodeStoreBase::dump(Formatter *f) const
 {
+  f->open_object_section("inode");
   inode.dump(f);
+  f->close_section();
   f->dump_string("symlink", symlink);
+
+  f->open_array_section("xattrs");
+  for (const auto &p : xattrs) {
+    f->open_object_section("xattr");
+    f->dump_string("key", p.first);
+    std::string v(p.second.c_str(), p.second.length());
+    f->dump_string("val", v);
```

Comment:
Why not `f->dump_string("val", val)`?

Reply:
type of xattr is compact_map<string, bufferptr>, os I have to convert bufferptr to string.

Reply:
k